scripts/get_ocp_addon_versions.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+import http.client
+import json
+import os
+import sys
+from urllib.parse import urlparse
+
+
+def parse_input():
+    """
+    Reads JSON input from stdin and parses it into a dictionary.
+    Returns:
+        dict: Parsed input data.
+    """
+    try:
+        data = json.loads(sys.stdin.read())
+    except json.JSONDecodeError as e:
+        raise ValueError("Invalid JSON input") from e
+    return data
+
+
+def validate_inputs(data):
+    """
+    Validates required inputs 'IAM_TOKEN' and 'REGION' from the parsed input.
+    Args:
+        data (dict): Input data parsed from JSON.
+    Returns:
+        tuple: A tuple containing (IAM_TOKEN, REGION).
+    """
+    token = data.get("IAM_TOKEN")
+    if not token:
+        raise ValueError("IAM_TOKEN is required")
+
+    region = data.get("REGION")
+    if not region:
+        raise ValueError("REGION is required")
+
+    return token, region
+
+
+def get_env_variable():
+    """
+    Retrieves the value of an environment variable.
+    Returns:
+        str: The value of the environment variable.
+    """
+    api_endpoint = os.getenv(
+        "IBMCLOUD_CS_API_ENDPOINT", "https://containers.test.cloud.ibm.com/global"
+    )
+    return api_endpoint
+
+
+def fetch_addon_versions(iam_token, region, api_endpoint):
+    """
+    Fetches openshift add-on versions using HTTP connection.
+    Args:
+        iam_token (str): IBM Cloud IAM token for authentication.
+        region (str): Region to query for add-ons.
+        api_endpoint (str): Base API endpoint URL.
+    Returns:
+        list: Parsed JSON response containing add-on information.
+    """
+    # Add https if user passed just a hostname
+    if not api_endpoint.startswith("https://"):
+        api_endpoint = f"https://{api_endpoint}"
+
+    parsed = urlparse(api_endpoint)
+
+    # Default path to /global if none supplied
+    base_path = parsed.path.rstrip("/") if parsed.path else "/global"
+
+    # Final API path
+    url = f"{base_path}/v1/addons"
+    host = parsed.hostname
+    headers = {
+        "Authorization": f"Bearer {iam_token}",
+        "Accept": "application/json",
+        "X-Region": region,
+    }
+
+    conn = http.client.HTTPSConnection(host)
+    try:
+        conn.request("GET", url, headers=headers)
+        response = conn.getresponse()
+        data = response.read().decode()
+
+        if response.status != 200:
+            raise RuntimeError(
+                f"API request failed: {response.status} {response.reason} - {data}"
+            )
+
+        return json.loads(data)
+    except http.client.HTTPException as e:
+        raise RuntimeError("HTTP request failed") from e
+    finally:
+        conn.close()
+
+
+def transform_addons(addons_data):
```

Comment:
This can be renamed along with description as this is dedicated to Cluster specific addons.